### PR TITLE
Yifan-Happy-Thought-Post

### PR DESCRIPTION
**Pull request by Yifan-858**
## Netlify link
https://yifan-happy-thought-post.netlify.app/

## Collaborators
[yifan-858]

### CODE REVIEW

File: src/component/post-section/PostSection.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    localStorage.getItem("shared-heart-count")
+      ? parseInt(localStorage.getItem("shared-heart-count"), 10)
+      : "0"
+  );
+
+  const handlePlusHeartCount = () => {
+    console.log("dependency rendered");
+    const plusHeartCount = localStorage.getItem("shared-heart-count")
+      ? parseInt(localStorage.getItem("shared-heart-count"), 10) + 1
+      : "1";
```
Comment (trista1987) [inline src/component/post-section/PostSection.jsx:16]: do we alwayt put the number 10? or it depends on the situation

File: src/component/post-time/PostTime.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    } else if (seconds === 1) {
+      displayTimeAgo = `${seconds} second ago`;
+    } else {
+      displayTimeAgo = `${seconds} seconds ago`;
+    }
+
+    setTimeago(displayTimeAgo);
+  }, [post.createdAt]);
+
+  return <p className="post-time">{timeAgo}</p>;
```
Comment (trista1987) [inline src/component/post-time/PostTime.jsx:39]: for the time, you can also use ```
const timeFormat = date => {
        const currentTime = new Date (date)
        return moment (currentTime).fromNow()
      }
```

**Review by trista1987 — COMMENTED** [review]
Very impressive. I learned SCSS and use localStorage from your code. Like the way how did you work on liked button and message length. Good job!

**Review by dzc1 — APPROVED** [review]
Amazing work Yifan, loving to see the destructuring of logic + comps on your 1App.jsx` file as well as your subtle animation upon creating a new thought.

Keep up the great work!

File: src/component/heart-display/HeartDisplay.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from "react";
+
+export const HeartDisplay = ({ post, handleUpdate, handlePlusHeartCount }) => {
+  const [isLiked, setIsLiked] = useState(
+    localStorage.getItem(`liked-${post._id}`)
+  );
```
[the remark this answers is not in the document]
Comment (Yifan-858) [reply src/component/heart-display/HeartDisplay.jsx:6]: Hi Trista!

I will try to explain as well as I can based on what I read for this project!

`liked-${post._id}`, this whole piece, is the key for the key-value pair you set for the localStorage object. 'liked - ' is a part of the key name. And '${post._id}' makes it more dynamic.

For example, the localStorage you saved could look like this: 'liked-12321231(whatever id): true'. You can also name it as you prefer, for example `clicked-${post._id}` or `heartBtn-${post._id}`. 

Once you successfully set your value to the localStorage by using localStorage.setItem(), you can check it in inspect -> Application -> Local storage -> click the down arrow -> click the URL, then you will see a list with the key-value pairs.

File: src/component/input-section/InputSection.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ message: newPost }),
+      });
+
+      if (!response.ok) {
+        console.log("Post failed");
```
[the remark this answers is not in the document]
Comment (Yifan-858) [reply src/component/input-section/InputSection.jsx:44]: Seems like 
"if (!response.ok) {
    throw new Error("Post failed");
}" works as the same.

File: src/component/post-time/PostTime.jsx (hunk 2 above)
Comment (Yifan-858) [reply src/component/post-time/PostTime.jsx:39]: Cool! Thank you for the new method will look into it!

File: src/component/heart-display/HeartDisplay.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (trista1987) [reply src/component/heart-display/HeartDisplay.jsx:6]: Hey thank you! Now I understand everything, no idea why yesterday I was confused

**Pull request closed without merging**